feat: import cURL on paste

anwarulislam opened the pull request:
<!--
Thanks for creating this pull request 🤗

Please make sure that the pull request is limited to one type (docs, feature, etc.) and keep it as small as possible. You can open multiple prs instead of opening a huge one.
-->

<!-- If this pull request closes an issue, please mention the issue number below -->
Closes # <!-- Issue # here -->

### Description
<!-- Add a brief description of the pull request -->

I've updated the code which contains the feature: now you can copy a cURL from the network console of the browser or from anywhere and paste it to the URL field then it will automatically recognize the cURL format. And will generate the request.

<!-- You can also choose to add a list of changes and if they have been completed or not by using the markdown to-do list syntax
- [x] Completed
-->

### Checks
<!-- Make sure your pull request passes the CI checks and do check the following fields as needed - -->
- [x] My pull request adheres to the code style of this project
- [x] All the tests have passed

### Additional Information
<!-- Any additional information like breaking changes, dependencies added, screenshots, comparisons between new and old behavior, etc. -->

liyasthomas commented:
cc: @AndrewBastin

AndrewBastin reviewed (changes requested):
@anwarulislam It's a really neat idea!

But, I am not exactly a big fan of the implementation, it would also be cool if we could preserve the line breaks as well (if the curl paste has multiple lines, it becomes a single line). in the Import Curl modal Also it would be cool if the pasted curl string doesn't appear on the URL field.

Would love your thoughts.

anwarulislam commented on packages/hoppscotch-app/components/http/ImportCurl.vue:
```diff
@@ -43,7 +43,9 @@ const curl = ref("")
-useCodemirror(curlEditor, curl, {
+const props = defineProps<{ show: boolean; text: string }>()
+
+const codeMirror = useCodemirror(curlEditor, curl, {
```
Thanks for the review. I had tried to implement in a different way with the `codeMirror` variable. But, this is not needed now. I'm removing it.

anwarulislam commented on packages/hoppscotch-app/components/http/ImportCurl.vue:
```diff
@@ -53,7 +55,17 @@ useCodemirror(curlEditor, curl, {
 })
 
-defineProps<{ show: boolean }>()
+watch(
+  () => props.show,
+  () => {
+    if (props.show) {
+      setTimeout(() => {
+        curl.value = props.text || ""
+      }, 100)
+    }
+  },
+  { immediate: false }
+)
```
Yeah, that's good. But, the problem is props.text containing previous value instead of updated value. that's why I added `setTimout` so that props get updated meanwhile. I think there must be a better way to do this. Could you please suggest me how can I accomplish this?

anwarulislam commented:
@AndrewBastin I've changed these things from the last review:

- using curlText variable for passing data to ImportCurl.vue component as prop.
- removed setTimout
- making URL field empty if it's curl after pasting
- and also supports multiline text. make sure you copied the text as multiline

AndrewBastin commented:
yup, can see that. reviewing it right now 🤣
